### PR TITLE
docs(activity-stream): add contributing.md

### LINKED ISSUE #1329: Add  contributing.md

### PR DESCRIPTION
**Pull request by tspurway**
fixes #1329
r? @sarracini

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# Contributing to the Activity Stream  Firefox addon
+
+Activity Stream is an experiment to enhance the functionality of Firefox's about:newtab page.  We welcome new 'streamers' to contribute to the project!
+
+- IRC: `#activity-stream` on `irc.mozilla.org`
+- Mailing List: [activity-stream-dev](https://groups.google.com/a/mozilla.com/d/forum/activity-stream-dev)
+
+## Architecture ##
+
+Activity stream is a Firefox addon.  It uses the Firefox [Addon SDK](https://developer.mozilla.org/en-US/Add-ons/SDK) and
```
Comment (pdehaan) [inline contributing.md]: Stream (uppercase S)?

File: contributing.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+is written in Javascript.  One of the cool things about Activity Stream is that the
```
Comment (pdehaan) [inline contributing.md]: JavaScript

File: contributing.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
+# Contributing to the Activity Stream  Firefox addon
```
Comment (pdehaan) [inline contributing.md]: Global: addon should be hyphenated (add-on), per the README in this repo, and AMO.

File: contributing.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+is written using [ReactJS](https://facebook.github.io/react/).  This makes it an awesome project for React hackers to contribute to!
+
+## Finding Bugs, Filing Tickets, Earning Karma ##
+
+Activity Stream lives on [GitHub](https://github.com/mozilla/activity-stream), but you already knew that!  If you've found
+a bug, or have a feature idea that you you'd like to see in Activity Stream, follow these simple guidelines:
+- Pick a thoughtful and terse title for the issue (ie. *not* Thing Doesn't Work!)
+- Make sure to mention your Firefox version, OS and basic system parameters (eg. Firefox 49.0, Windows XP, 512KB RAM)
+- If you can reproduce the bug, give a step-by-step recipe
+- You probably don't want to tag the bug, but if you must, pick either `Bug` or `Feature`
```
Comment (pdehaan) [inline contributing.md]: Not sure that casual users can add tags, only contributors/admins.

File: contributing.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+All work on Activity Stream is broken into two week iterations, which we map into a GitHub [Milestone](https://github.com/mozilla/activity-stream/milestones).  At the beginning of the iteration, we prioritize and estimate tickets
+into the milestone, attempting to figure out how much progress we can make during the iteration.  
+
+## Priorities, Comments ##
+
+All tickets that have been [triaged](#triage) will have a priority tag of either `P1`, `P2`, or `P3`, which are highest to lowest
+priorities of tickets in Activity Stream.  We love ticket tags and you might also see `Blocked`, `Critical` or `Chemspill` tags, which
+indicate our level of anxiety about that particular ticket.  
+
+If you are new to the repo, you might want to pay close attention to `Good first bug`, `Bug`, `Chore` and `Polish` tags, as these are
```
Comment (pdehaan) [inline contributing.md]: We may want to link these to GitHub issues. For example:

- "Good first bug" => https://github.com/mozilla/activity-stream/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+first+bug%22
- "Bug" => https://github.com/mozilla/activity-stream/issues?q=is%3Aopen%20is%3Aissue%20label%3ABug%20
- "Chore" => https://github.com/mozilla/activity-stream/issues?q=is%3Aopen+is%3Aissue+label%3AChore
- "Polish" => https://github.com/mozilla/activity-stream/issues?q=is%3Aopen+is%3Aissue+label%3APolish

File: contributing.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+ satisfied that your code is good-to-go, you will get the coveted `R+` comment, and your code can be merged.  If you have
+ commit permission, you can go ahead and merge the code to `master`, otherwise, it will be done for you.
+ 
+Our project prides itself on it's respectful, patient and positive attitude when it comes to reviewing contributor's code, and as such,
+we expect contributors to be respectful, patient and positive in their communications as well.  Let's be friends and learn
+from each other for a free and awesome web!
+
+[Mozilla Committing Rules and Responsibilities](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Committing_Rules_and_Responsibilities)
+
+## Unit Tests ##
```
Comment (pdehaan) [inline contributing.md]: Missing content?

Comment (tspurway) [inline contributing.md]: I don't feel qualified to describe our unit testing practice.  Anyone else want to take a crack at it?

Comment (sarracini) [inline contributing.md]: Kate wrote a doc about how to write addon tests (https://github.com/mozilla/activity-stream/blob/master/content-test/addon-tests-how-to.md). It's pretty detailed, so maybe just point them to that doc, and just mention here the ```npm run test``` command to run the entire test suite (and to make sure they all pass and coverage is good). Also maybe just mention that all tests go in ```content-test/```

**Comment by sarracini** [comment]
Looks good, I dropped a comment about the unit test section, so it's R+ with that addition

**Pull request merged**